packages/blockchain-sdk-solana/src/auth/solana-authentication-facade-factory.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  AuthenticationFacadeFactory,
+  Config,
+  TokenHeaderParser,
+  TokenValidator,
+} from '@dialectlabs/sdk';
+import type { DialectSolanaWalletAdapterWrapper } from '../wallet-adapter/dialect-solana-wallet-adapter-wrapper';
+import { SolanaEd25519AuthenticationFacadeFactory } from './ed25519/solana-ed25519-authentication-facade-factory';
+import { DialectWalletAdapterSolanaEd25519TokenSigner } from './ed25519/solana-ed25519-token-signer';
+import { SolanaEd25519TokenValidator } from './ed25519/solana-ed25519-token-validator';
+import { SolanaTxAuthenticationFacadeFactory } from './tx/solana-tx-authentication-facade-factory';
+import { DialectWalletAdapterSolanaTxTokenSigner } from './tx/solana-tx-token-signer';
+import { SolanaTxTokenValidator } from './tx/solana-tx-token-validator';
+
+export class SolanaAuthenticationFacadeFactory {
+  solanaEd25519TokenValidator: TokenValidator;
+
+  solanaTxTokenValidator: TokenValidator;
+
+  constructor(
+    readonly config: Config,
+    readonly wallet: DialectSolanaWalletAdapterWrapper,
+  ) {
+    this.solanaEd25519TokenValidator = new SolanaEd25519TokenValidator();
+    this.solanaTxTokenValidator = new SolanaTxTokenValidator();
+  }
+
+  create(): AuthenticationFacadeFactory {
+    const tokenHeader = this.getCachedTokenHeader();
+    if (
```

Comment:
There's pretty complex logical expression here with high cyclomatic complexity and I def feel there're corner cases in this solution because of fallback to `SolanaEd25519AuthenticationFacadeFactory` and  the fact that single factory is used in runtime after initialisation => bugs can happen e.g. if you have cached token, initialise specific factory, but then token expires in runtime

If you wanna proceed in using this approach, I would move the decorator from factory to facade since you may need to deal with both facades in runtime before and after token expires